content/advanced/340_appmesh_flagger/canary_deploy.md

```diff
@@ -0,0 +1,201 @@
+---
+title: "Deploy Canary Set Up"
+date: 2021-03-18T00:00:00-05:00
+weight: 30
+draft: false
+---
+
+#### Clone the repository
+
+```
+cd ~/environment
+git clone https://github.com/aws-containers/eks-microservice-demo.git
+cd eks-microservice-demo
+```
+
+#### Build and Push the container images
+Here we are building and pushing three versions of `detail` backend service container images and one version of `frontend` service container image to [Amazon ECR](https://aws.amazon.com/ecr/). Once completed, you can confirm the images are in ECR by logging into the console.
+
+
+```bash
+aws ecr get-login-password --region $AWS_REGION | docker login --username AWS --password-stdin $ACCOUNT_ID.dkr.ecr.$AWS_REGION.amazonaws.com
+PROJECT_NAME=eks-microservice-demo
+export APP_VERSION=1.0
+export APP_VERSION_2=2.0
+export APP_VERSION_3=3.0
+for app in detail; do
+  aws ecr describe-repositories --repository-name $PROJECT_NAME/$app >/dev/null 2>&1 || \
+  aws ecr create-repository --repository-name $PROJECT_NAME/$app >/dev/null
+  TARGET=$ACCOUNT_ID.dkr.ecr.$AWS_REGION.amazonaws.com/$PROJECT_NAME/$app:$APP_VERSION
+  docker build -t $TARGET apps/$app
+  docker push $TARGET
+  cd apps/$app
+  TARGET2=$ACCOUNT_ID.dkr.ecr.$AWS_REGION.amazonaws.com/$PROJECT_NAME/$app:$APP_VERSION_2
+  docker build -t $TARGET2 -f version2/Dockerfile .
+  docker push $TARGET2
+  TARGET3=$ACCOUNT_ID.dkr.ecr.$AWS_REGION.amazonaws.com/$PROJECT_NAME/$app:$APP_VERSION_3
+  docker build -t $TARGET3 -f version3/Dockerfile .
+  docker push $TARGET3
+  cd ../../.
+done
+for app in frontend; do
+  aws ecr describe-repositories --repository-name $PROJECT_NAME/$app >/dev/null 2>&1 || \
+  aws ecr create-repository --repository-name $PROJECT_NAME/$app >/dev/null
+  TARGET=$ACCOUNT_ID.dkr.ecr.$AWS_REGION.amazonaws.com/$PROJECT_NAME/$app:$APP_VERSION
+  docker build -t $TARGET apps/$app
+  docker push $TARGET
+done
+
+```
+{{% notice info %}}
+Building/Pushing Container images first time to ECR may take around 3-5 minutes
+{{% /notice %}}
+
+#### Create the namespace and mesh
```

Comment:
Can we change this subtitle to "Create a namespace and a mesh".

Reply:
done